File: sorting/bubbleSort.js
function swap(items, firstIndex, secondIndex){
	var temp = items[firstIndex];
	items[firstIndex] = items[secondIndex];
	items[secondIndex] = temp;
}

function bubbleSort(items){
	var len = items.length,
		i, j, stop;

	for (i=0; i < len; i++){
		for (j=0, stop=len-i; j < stop; j++){
			if (items[j] > items[j+1]){
				swap(items, j, j+1);
			}
		}
	}
	return items;
}

var array = [1, 4, 3, 18, 9, 10, 2];

var sortedArray = bubbleSort(array);

console.log(sortedArray)

File: sorting/bubble-sort.rb
def swap(items, first_i, second_i)
	placeholder = items[first_i]
	items[first_i] = items[second_i]
	items[second_i] = placeholder
end

def bubble_sort(items)
	return list if list.size <= 1 # already sorted
	swapped = true

	while swapped do
		swapped = false
		0.upto(list.size - 2) do |i|
			if list[i] > list[i+1]
				list[i], list[i+1] = list[i+1], list[i] #swap values
				swapped = true
			end
		end
	end
end